src/app/(frontend)/[...slug]/page.tsx

```diff
@@ -57,6 +58,16 @@ export default async function Page({ params: paramsPromise }: Args) {
   const { slug = ['home'] } = await paramsPromise;
   const url = '/' + slug.join('/');
 
+  // **Check if "event" is in the URL**
+
+  const eventIndex = slug.findIndex((segment) => segment === 'event');
+  const eventId = eventIndex !== -1 ? slug[eventIndex + 1] : null;
+
+  // **If "event" is in the URL and has an ID, render EventPage instead**
+  if (eventId) {
+    return <EventPage id={eventId} />;
+  }
+
```

Comment:
I would prefer to have a custom `src/app/event/[eventId]/page.ts` file over this branching code.